src/renderer/components/command-palette/command-container.tsx

```diff
@@ -34,47 +35,38 @@ interface Dependencies {
 
 @observer
 class NonInjectedCommandContainer extends React.Component<Dependencies> {
-  private escHandler = (event: KeyboardEvent) => {
-    if (event.key === "Escape") {
-      event.stopPropagation();
-      this.props.commandOverlay.close();
-    }
-  };
-
-  handleCommandPalette = () => {
-    const matchedClusterId = this.props.matchedClusterId.get();
-
-    if (matchedClusterId !== undefined) {
-      broadcastMessage(`command-palette:${matchedClusterId}:open`);
-    } else {
-      this.props.commandOverlay.open(<CommandDialog />);
-    }
-  };
-
-  onKeyboardShortcut(action: () => void) {
-    return ({ key, shiftKey, ctrlKey, altKey, metaKey }: KeyboardEvent) => {
-      const ctrlOrCmd = this.props.isMac ? metaKey && !ctrlKey : !metaKey && ctrlKey;
-
-      if (key === "p" && shiftKey && ctrlOrCmd && !altKey) {
-        action();
-      }
-    };
-  }
-
   componentDidMount() {
-    const { clusterId, addWindowEventListener, commandOverlay } = this.props;
+    const { clusterId, addWindowEventListener, commandOverlay, matchedClusterId, isMac } = this.props;
 
     const action = clusterId
       ? () => commandOverlay.open(<CommandDialog />)
-      : this.handleCommandPalette;
+      : () => {
+        const matchedId = matchedClusterId.get();
+
+        if (matchedId) {
+          broadcastMessage(`command-palette:${matchedClusterId}:open`);
+        } else {
+          commandOverlay.open(<CommandDialog />);
+        }
+      };
     const ipcChannel = clusterId
       ? `command-palette:${clusterId}:open`
       : "command-palette:open";
 
     disposeOnUnmount(this, [
       this.props.legacyOnChannelListen(ipcChannel, action),
-      addWindowEventListener("keydown", this.onKeyboardShortcut(action)),
-      addWindowEventListener("keyup", this.escHandler, true),
+      addWindowEventListener("keydown", onKeyboardShortcut(
+        isMac
+          ? "Shift+Cmd+P"
+          : "Shift+Ctrl+P",
+        action,
+      )),
```

Comment:
All this mount-time stuff would be removed from here if we had the keyboard-shortcuts -feature. 

Just saying because normally I would say that this is logic that is not welcome here at all, but for now I think it's OK because we shouldn't introduce the keyboard-shortcuts feature here :)